homeassistant/components/climate/venstar.py

```diff
@@ -0,0 +1,287 @@
+"""
+Support for Venstar WiFi Thermostats.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.venstar/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    STATE_COOL, STATE_HEAT, STATE_IDLE, STATE_AUTO,
+    ClimateDevice, PLATFORM_SCHEMA,
+    SUPPORT_TARGET_TEMPERATURE, SUPPORT_TARGET_HUMIDITY,
+    SUPPORT_FAN_MODE, SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE_HIGH,
+    SUPPORT_TARGET_TEMPERATURE_LOW,
+    ATTR_TARGET_TEMP_HIGH, ATTR_TARGET_TEMP_LOW)
+
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_SSL, CONF_USERNAME, CONF_TIMEOUT,
+    TEMP_FAHRENHEIT, ATTR_TEMPERATURE, TEMP_CELSIUS,
+    PRECISION_WHOLE, STATE_ON)
+
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['venstarcolortouch==0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_SSL = False
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_USERNAME, default=None): cv.string,
+    vol.Optional(CONF_PASSWORD, default=None): cv.string,
+    vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+    vol.Optional(CONF_TIMEOUT, default=5):
+        vol.All(vol.Coerce(int), vol.Range(min=1))
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Venstar thermostat."""
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    host = config.get(CONF_HOST)
+    if config.get(CONF_SSL):
+        proto = 'https'
+    else:
+        proto = 'http'
+    timeout = config.get(CONF_TIMEOUT)
+
+    import venstarcolortouch
+    client = venstarcolortouch.VenstarColorTouch(addr=host,
+                                                 timeout=timeout,
+                                                 user=username,
+                                                 password=password,
+                                                 proto=proto)
+
+    add_devices([VenstarThermostat(client)])
```

Comment:
If you call this with `True` as second parameter, it will schedule an update right after adding the device. That might be desirable?